backend/app/views/spree/admin/promotion_code_batches/_form_fields.html.erb

```diff
@@ -0,0 +1,22 @@
+<div class="field">
+  <%= batch.label :base_code, class: "required" %>
+  <%= batch.text_field :base_code, class: "fullwidth", required: true %>
+</div>
+<div class="field">
+  <%= batch.label :number_of_codes, class: "required" %>
+  <%= batch.number_field :number_of_codes, class: "fullwidth", min: 1, required: true %>
+</div>
+<div class="field">
+  <%= batch.label :join_characters %>
+  <%= batch.text_field :join_characters, class: "fullwidth" %>
+</div>
+<% unless params["promotion_id"] %>
```

Comment:
Instead of relying on global state, perhaps we can pass the promotion ID (or just a boolean flag) to the partial as a local variable? This would make the partial more self-contained.

Reply:
Updated partials to pass local variable promotion_id